test/core/security/grpc_tls_crl_provider_test.cc

```diff
@@ -83,6 +101,110 @@ TEST(CrlProviderTest, StaticCrlProviderLookupBad) {
   EXPECT_EQ(crl, nullptr);
 }
 
+TEST(CrlProviderTest, DirectoryReloaderCrlLookupGood) {
+  auto result = experimental::CreateDirectoryReloaderProvider(
+      kCrlDirectory, std::chrono::seconds(1), nullptr);
+  ASSERT_TRUE(result.ok());
+  std::shared_ptr<CrlProvider> provider = std::move(*result);
+
+  CertificateInfoImpl cert = CertificateInfoImpl(kCrlIssuer);
+  auto crl = provider->GetCrl(cert);
+  ASSERT_NE(crl, nullptr);
+  ASSERT_EQ(crl->Issuer(), kCrlIssuer);
+
+  CertificateInfoImpl intermediate =
+      CertificateInfoImpl(kCrlIntermediateIssuer);
+  auto intermediate_crl = provider->GetCrl(intermediate);
+  ASSERT_NE(intermediate_crl, nullptr);
+  ASSERT_EQ(intermediate_crl->Issuer(), kCrlIntermediateIssuer);
+}
+
+TEST(CrlProviderTest, DirectoryReloaderCrlLookupBad) {
+  auto result = experimental::CreateDirectoryReloaderProvider(
+      kCrlDirectory, std::chrono::seconds(1), nullptr);
+  ASSERT_TRUE(result.ok());
+  std::shared_ptr<CrlProvider> provider = std::move(*result);
+
+  CertificateInfoImpl bad_cert = CertificateInfoImpl("BAD CERT");
+  auto crl = provider->GetCrl(bad_cert);
+  ASSERT_EQ(crl, nullptr);
+}
+
+std::string MakeTempDir() {
+  char templ[] = "/tmp/tmpdir.XXXXXX";
+  std::string dir_path = mkdtemp(templ);
+  return dir_path;
+}
+
+std::string TempDirNameFromPath(absl::string_view dir_path) {
+  std::vector<std::string> split = absl::StrSplit(dir_path, "/");
+  return split[2] + "/";
```

Comment:
Hard-coding element 2 seems brittle.  How do we know that's always the right one?  Seems like what you really want here is to find the last slash and return everything after that.

Maybe what you're after here is the libc `basename()` function?  But note the same question as above w.r.t. portability.

Reply:
We know this always follows from `char directory_template[] = "/tmp/tmpdir.XXXXXX";`, so we should be guaranteed here (the `mkdtemp` call replaces `XXXXXX` in such a way that it makes the path unique).